components/notebook-controller/e2e/suite_test.go

```diff
@@ -0,0 +1,231 @@
+	err = applyKustomizations()
+	Expect(err).NotTo(HaveOccurred())
+
+	// +kubebuilder:scaffold:scheme
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sClient).ToNot(BeNil())
+
+	return nil
+
+}, func(data []byte) {
+})
+
+// Cleaning the ENV
+var _ = SynchronizedAfterSuite(func() {
+}, func() {
+	if !useExistingCluster {
+		By("Tearing down the test k8s cluster created by the test.")
+		Expect(deleteK3DCluster(clusterName)).Should(BeNil())
+	} else {
+		By("Deleting the deployed CRD's in the existing cluster.")
+		// Deleting CRD's that were deployed as part of the test.
+		err := deleteKustomizations()
+		Expect(err).Should(BeNil())
+	}
+})
+
+// bootstrapk8sCluster creates a k3d cluster with a random name. Loads the docker image into the cluster.
+// The docker image name retrieved from the ENV variable TestImage.
+func bootstrapk8sCluster() error {
+	clusterName = fmt.Sprintf("e2e-notebook-kf-cluster-%s", randStringRunes(6))
+
+	cmd := exec.Command("k3d", "cluster", "create", clusterName, "-i", k3DockerImage,
+		"--no-lb", "--wait", "--timeout", "5m",
+		"--update-default-kubeconfig=false")
+	cmd.Stderr, cmd.Stdout = os.Stderr, os.Stdout
+	err := cmd.Run()
+
+	if err != nil {
+		return err
+	}
+	cmd = exec.Command("k3d", "kubeconfig", "write", clusterName)
+	err = cmd.Run()
+
+	if err != nil {
+		return err
+	}
+	kubeconfigPath := path.Join(os.Getenv("HOME"), fmt.Sprintf(".k3d/kubeconfig-%s.yaml", clusterName))
+	_ = os.Setenv("KUBECONFIG", kubeconfigPath)
+	imageName := os.Getenv(TestImage)
+	if len(imageName) == 0 {
+		return errors.New("the env variable TEST_IMAGE is not set to load the image into the k3d cluster")
+	}
+	cmd = exec.Command("k3d", "image", "import", "-c", clusterName, imageName)
+	cmd.Stderr, cmd.Stdout = os.Stderr, os.Stdout
+	return cmd.Run()
+}
+
+// deleteK3DCluster deletes the dynamically created k3d cluster.
+func deleteK3DCluster(clusterName string) error {
+	return exec.Command("k3d", "cluster", "delete", clusterName).Run()
+}
+
+// applyKustomizations applies kustomizations into the k8s clusters.
+func applyKustomizations() error {
+	kubectlApply := exec.Command("kubectl", "apply", "-f",
+		filepath.Join("..", "config", "crd", "bases"))
+	kubectlApply.Stdout, kubectlApply.Stderr = os.Stdout, os.Stderr
+	err := kubectlApply.Run()
+
+	if err != nil {
+		return err
+	}
+
+	out, err := exec.Command("kustomize", "build",
+		filepath.Join("..", "config", "default")).Output()
+
+	if err != nil {
+		return err
+	}
+
+	kubectlApply = exec.Command("kubectl", "apply", "-f", "-")
+	kubectlApply.Stdin = bytes.NewReader(out)
+	kubectlApply.Stdout, kubectlApply.Stderr = os.Stdout, os.Stderr
+	err = kubectlApply.Run()
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deleteKustomizations deletes the CRD,RBAC,Deployment and services of the notebook controller from the cluster.
+func deleteKustomizations() error {
```

Comment:
Should we make deleteKustomizations take a list of paths as input, so it's re-usable?